access to the license server.

enescantuerk opened the pull request:
hi

If there is no access to the license server, it gives a license error. It would be great if this solved!

(No action taken if there is no access to the server)

michelve committed:
Set theme jekyll-theme-cayman

michelve committed:
Fixes an issue where license type will not shown when creating a license product in woocommerce

michelve committed:
Fixes an issue where license type will not shown when creating a license product in woocommerce

michelve committed:
Set theme jekyll-theme-minimal

michelve committed:
Fix an issue that prevented the admin dashboard from opening when woo-commerce was not installed or active

michelve committed:
Fix an issue that prevented the admin dashboard from opening when woo-commerce was not installed or active

michelve committed:
Fix an issue that prevented the admin dashboard from opening when woo-commerce was not installed or active

michelve committed:
Fix an issue that prevented the admin dashboard from opening when woo-commerce was not installed or active

michelve committed:
Updated change-log

dependabot[bot] committed:
Bump handlebars from 4.1.2 to 4.5.3

Bumps [handlebars](https://github.com/wycats/handlebars.js) from 4.1.2 to 4.5.3.
- [Release notes](https://github.com/wycats/handlebars.js/releases)
- [Changelog](https://github.com/wycats/handlebars.js/blob/master/release-notes.md)
- [Commits](https://github.com/wycats/handlebars.js/compare/v4.1.2...v4.5.3)

Signed-off-by: dependabot[bot] <support@github.com>

michelve committed:
Merge pull request #12 from michelve/dependabot/npm_and_yarn/handlebars-4.5.3

Bump handlebars from 4.1.2 to 4.5.3

michelve committed:
5.1.8
--------
Updated license overview page
Fixed minor php warnings
Added about page
Added license key to file name when exporting a license key

michelve committed:
Merge branch 'master' of https://github.com/michelve/software-license-manager

michelve committed:
updated readme and change-log files

michelve committed:
updated readme and change-log files

michelve committed:
Merge branch 'master' of https://github.com/michelve/software-license-manager

michelve committed:
- Users with blank $wc_billing_email can see all other licenses

michelve committed:
5.2.0 - License overview was not displaying list of activated domains

michelve committed:
5.2.0 release

MechComp committed:
Sorting by ID not working properly

When sorting by ID and comparing strings, the order is wrong - it sorts by the first letter in the string - example: descending order 99,98,97...100

MechComp committed:
Error - sets all licences as Expired if they don't have Expiration date set

After upgrading SLM from old version all my licences were set as expired even though they don't have expiration date set.
Also the function check_for_expired_lic sends e-mail to the Customer and creates log entry even though the Auto expiration is not set.

MechComp committed:
Sorting by ID not working properly

michelve committed:
Merge pull request #17 from MechComp/patch-2

Error - sets all licences as Expired if they don't have Expiration da…

michelve committed:
Merge pull request #18 from MechComp/patch-1

Sorting by ID not working properly

michelve committed:
Merge pull request #19 from MechComp/master

Sorting by ID not working properly

michelve committed:
Error - sets all licenses as Expired if they don't have Expiration date
Sorting by ID not working properly
thanks to @MechComp for the help

michelve committed:
5.2.1 changelog

michelve committed:
Fixed: License type was not showing correctly when editing a license
Fixed: Activation date was not displaying correctly when saving or adding new licenses
UPDATE: when a license is set to lifetime the expiration field will be disabled and ignore.
FIXED: when a license was set to lifetime the expiration was getting populated with a future date.
Improvement: fixed some typos and improved the Spanish language pack
Updated: updated the product type template to work with any theme.

michelve committed:
no message

michelve committed:
5.3.2 - make lifetime license work better with woocommerce download functionality

michelve committed:
IMPROVED: removed unnecessary code
Fixed: license was not showing registered domains or devices in my account page

michelve committed:
pot language file added

michelve committed:
Added: new icons to manage license interface.
Improved: export functionality in manage license page
Added: activation email - the user will get notified once the license key is activated
Added: icon thumbnails to api response (for domain and devices)
Added: slm_info action to api (when called it only returns license information it ignores domains and devices)
Added: Conflict mode tweak (if un-checked it will remove the bootstrap library)

Signed-off-by: Michel Velis <michel@epikly.com>

michelve committed:
added: support for item reference (now you can tag a license under plugins, themes, code, etc), for previous version, a default tag will be set, it is recommended to update the items in your store/or licenses and change the item-reference to match your scenario.
fixed: fix a bug where db options would get reset after applying a new update.

Signed-off-by: Michel Velis <michel@epikly.com>

michelve committed:
Added: auto updater support

Signed-off-by: Michel Velis <michel@epikly.com>

michelve committed:
SML now support auto updates.

Signed-off-by: Michel Velis <michel@epikly.com>

michelve committed:
updated wp readme

Signed-off-by: Michel Velis <michel@epikly.com>

MechComp committed:
Merge pull request #1 from michelve/master

Sync - 2020-04-29

michelve committed:
added: validation when creating a new license using the admin dashboard
improved: item reference (there were cases where it will return empty values)
improved: woocommerce support, now slm auto auto populates values from the settings panels when creating a new license product
improved: toggling off slm admin bar removes the shortcut link from the admin bar
improved: language internalization

Signed-off-by: Michel Velis <michel@epikly.com>

michelve committed:
Set theme jekyll-theme-cayman

michelve committed:
5.5.2 release

michelve committed:
updated image preview

michelve committed:
updated image preview

MechComp committed:
Merge pull request #3 from michelve/master

Sync 2020-05-07

MechComp committed:
Merge pull request #4 from MechComp/master

Sync 2020-05-07

michelve committed:
Updated changelog

michelve committed:
updated TODO list

michelve committed:
5.5.3 - Fixed a product editing glitch with the WP eStore plugin integration.

michelve committed:
5.5.3 - Fixed a product editing glitch with the WP eStore plugin integration.

MechComp committed:
Merge pull request #5 from michelve/master

Version 5.5.3 merge

MechComp committed:
Merge from Michelve - v 5.5.3

Merge from Michelve - v 5.5.3

MechComp committed:
5.5.4 - Stats and licences overview changes

MechComp committed:
5.5.4 - fix for Item referece ignored on activation

MechComp committed:
5.5.4 - CSS update

MechComp committed:
5.5.4 - order meta fix, expiration not compulsory

MechComp committed:
5.5.4 - fix for Expired license date when empty in My licenses

michelve committed:
Merge pull request #24 from MechComp/MechComp

MechComp

MechComp committed:
5.5.4 - Orders backward compatibility view; fixed icon

michelve committed:
Merge pull request #25 from MechComp/MechComp

5.5.4 - Orders backward compatibility view; fixed icon

MechComp committed:
5.5.4 - bugfix - validation of licence fields - product editor

michelve committed:
Merge pull request #26 from MechComp/MechComp

5.5.4 - bugfix - validation of licence fields - product editor

michelve committed:
Update software-license-manager.php

michelve committed:
Update readme.txt

michelve committed:
Update README.md

MechComp committed:
5.5.5 - Order mail fix

michelve committed:
Merge pull request #28 from MechComp/MechComp

5.5.5 - Order mail fix

michelve committed:
Update README.md

michelve committed:
Update readme.txt

michelve committed:
Update software-license-manager.php

michelve committed:
Update software-license-manager.php

michelve committed:
Merge branch 'master' of https://github.com/michelve/software-license-manager

michelve committed:
no message

michelve committed:
5.7.0

michelve committed:
implemneted data-title support

michelve committed:
Merge pull request #34 from michelve/33-data-title-support

implemneted data-title support

michelve committed:
added new icons

michelve committed:
added new icons

added new icons

michelve committed:
resolved bootstrap issues and conflicts

resolved bootstrap issues and conflicts with woocommerce

michelve committed:
updated pot and mo language files

updated pot and mo language files

michelve committed:
5.5.7 release

michelve committed:
updated date() to wp_date()

michelve committed:
updated change-log and package.json

dependabot[bot] committed:
Bump node-fetch from 2.6.0 to 2.6.1

Bumps [node-fetch](https://github.com/bitinn/node-fetch) from 2.6.0 to 2.6.1.
- [Release notes](https://github.com/bitinn/node-fetch/releases)
- [Changelog](https://github.com/node-fetch/node-fetch/blob/master/docs/CHANGELOG.md)
- [Commits](https://github.com/bitinn/node-fetch/compare/v2.6.0...v2.6.1)

Signed-off-by: dependabot[bot] <support@github.com>

michelve committed:
Create SECURITY.md

michelve committed:
questions template added

michelve committed:
templates

michelve committed:
no message

michelve committed:
no message

michelve committed:
updated license

k-kikuchi-waverworks committed:
Update fixing bugs.

https://github.com/michelve/software-license-manager/issues/45
https://github.com/michelve/software-license-manager/issues/43
https://github.com/michelve/software-license-manager/issues/42
https://github.com/michelve/software-license-manager/issues/41

k-kikuchi-waverworks committed:
Merge pull request #1 from k-kikuchi-idea-hack/k-kikuchi-idea-hack-patch-1

Update fixing bugs.

michelve committed:
Merge pull request #46 from k-kikuchi-idea-hack/master

I created bug fix for wc_licenses_class.php

michelve committed:
Merge pull request #37 from michelve/dependabot/npm_and_yarn/node-fetch-2.6.1

Bump node-fetch from 2.6.0 to 2.6.1

michelve committed:
use of home_url instead of site_url

michelve committed:
version 5.5.9

michelve committed:
Merge branch 'master' of https://github.com/michelve/software-license-manager

michelve committed:
esc_html and home_url fixes

michelve committed:
added composer support

michelve committed:
fixes issue: cannot delete an actively licensed device from the license manager #47

MechComp committed:
Merge pull request #7 from michelve/master

Merge from parent

MechComp committed:
Bug-fix - multiple licences created when order item qty changed

michelve committed:
Merge pull request #50 from MechComp/master

Bug-fix - multiple licences created when order item qty changed

michelve committed:
test

michelve committed:
Fixes issue: Clear Logs saved #51

michelve committed:
change-log 5.5.10

michelve committed:
removed unneeded files fixes debug logger issue - Problem in the process of activating and deactivating user-side licenses #52

michelve committed:
fixes debug logger issue - Problem in the process of activating and deactivating user-side licenses #52

michelve committed:
Merge branch 'master' of https://github.com/michelve/software-license-manager

michelve committed:
removed files

michelve committed:
hide sample code

MechComp committed:
Merge pull request #8 from michelve/master

Sync with Michelve master

MechComp committed:
Front-end licence removal doesn't reflect setting allow_user_activation_removal

dependabot[bot] committed:
Bump handlebars from 4.5.3 to 4.7.7

Bumps [handlebars](https://github.com/wycats/handlebars.js) from 4.5.3 to 4.7.7.
- [Release notes](https://github.com/wycats/handlebars.js/releases)
- [Changelog](https://github.com/handlebars-lang/handlebars.js/blob/master/release-notes.md)
- [Commits](https://github.com/wycats/handlebars.js/compare/v4.5.3...v4.7.7)

Signed-off-by: dependabot[bot] <support@github.com>

michelve committed:
Merge pull request #59 from michelve/dependabot/npm_and_yarn/handlebars-4.7.7

Bump handlebars from 4.5.3 to 4.7.7

appthemes7 commented:
Hi, Thank you. This pull is too big with too many changes all in one so we won't be able to merge it. Pull request needs to be submitted with smaller changes at a time (so we can review the code changes easily). We can't take the risk of breaking any user's site with big changes.

enescanturkcom commented:
So can you create a sample code string?

for license check.

appthemes7 commented:
You mean something like the following to check a license key?
`
$api_params = array(
'slm_action' => 'slm_check',
'secret_key' => '5421048138b321.90068894',
'license_key' => 'KEYTOCHECK',
);
// Send query to the license manager server
$response = wp_remote_get(add_query_arg($api_params, YOUR_LICENSE_SERVER_URL), array('timeout' => 20, 'sslverify' => false));
`

enescanturkcom commented:
Yes.

I have software that I sell to my customers. I have to check their license.

I am looking for such a way that my customers will not be harmed if I close my site in the future.

my code:
`define('YOUR_LICENSE_SERVER_URL', 'https://wpdoz.com/'); 

	$api_params = array(
		'slm_action' => 'slm_check',

		'secret_key' => 'my_secret_key',

		'license_key' => get_option('sample_license_key'),

	);
	$response = wp_remote_get(add_query_arg($api_params, YOUR_LICENSE_SERVER_URL), array('timeout' => 20, 'sslverify' => false));
	$license_data = json_decode(wp_remote_retrieve_body($response));
	global $active, $message;
	if($license_data->result == 'success'){?>
	     
	<?php }else{ ?>
            <?php 
				
				if ( ! is_admin() ) {
				
				wp_die( __('not licensed!.') ); 
				}
?>

        <?php }

?> `

Commit:
Update slm_about_menu.php

Commit:
Update slm-list-licenses-class.php

Commit:
Update purchase.php

Commit:
Update slm-utility.php

Commit:
Update stats.php

Commit:
Update slm-add-licenses.php

Commit:
Update slm-subscribers.php

Commit:
Update slm-admin-functions.php

Commit:
Update slm-lic-settings.php

michelve committed:
Merge pull request #70 from mzoroufchi/patch-4

Update slm-list-licenses-class.php

michelve committed:
Merge pull request #69 from mzoroufchi/patch-3

Update slm_about_menu.php

michelve committed:
Merge pull request #68 from mzoroufchi/patch-7

Update stats.php

michelve committed:
Merge pull request #67 from mzoroufchi/patch-6

Update slm-utility.php

michelve committed:
Merge pull request #66 from mzoroufchi/patch-5

Update purchase.php

michelve committed:
Merge pull request #65 from mzoroufchi/patch-11

Update slm-lic-settings.php

michelve committed:
Merge pull request #64 from mzoroufchi/patch-10

Update slm-admin-functions.php

michelve committed:
Merge pull request #63 from mzoroufchi/patch-8

Update slm-add-licenses.php

michelve committed:
Merge pull request #62 from mzoroufchi/patch-9

Update slm-subscribers.php

Commit:
Update slm-add-licenses.php

michelve committed:
Merge pull request #71 from mzoroufchi/patch-12

Update slm-add-licenses.php

michelve committed:
Merge pull request #57 from MechComp/master

Front-end licence removal doesn't reflect setting allow_user_activation_removal

michelve committed:
Updated License Details view inside my account page > my licenses (woo-commerce support)
replaced expand view for modal for my licenses inside my licenses page
Update WP support

michelve committed:
5.6.1 release

michelve committed:
updated read me

michelve committed:
Update README.md

michelve committed:
WIP: Added a new action hook for when it receives an activation request for an expired license key (slm_api_listener_slm_activate_key_expired).

dependabot[bot] committed:
Bump node-fetch from 2.6.1 to 2.6.7

Bumps [node-fetch](https://github.com/node-fetch/node-fetch) from 2.6.1 to 2.6.7.
- [Release notes](https://github.com/node-fetch/node-fetch/releases)
- [Commits](https://github.com/node-fetch/node-fetch/compare/v2.6.1...v2.6.7)

---
updated-dependencies:
- dependency-name: node-fetch
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

michelve committed:
Merge pull request #80 from michelve/dependabot/npm_and_yarn/node-fetch-2.6.7

Bump node-fetch from 2.6.1 to 2.6.7

dependabot[bot] committed:
Bump minimist from 1.2.5 to 1.2.6

Bumps [minimist](https://github.com/substack/minimist) from 1.2.5 to 1.2.6.
- [Release notes](https://github.com/substack/minimist/releases)
- [Commits](https://github.com/substack/minimist/compare/1.2.5...1.2.6)

---
updated-dependencies:
- dependency-name: minimist
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

michelve committed:
Merge pull request #85 from michelve/dependabot/npm_and_yarn/minimist-1.2.6

Bump minimist from 1.2.5 to 1.2.6

Narimm committed:
Correct error on activate that trys to load a non-existent option.

(cherry picked from commit 70bfd1f2ed23582b2c3758e220f0f9d74fa2d16d)

michelve committed:
Merge pull request #90 from Narimm/fix-option-error

Correct error on activate that trys to load a non-existent option.

michelve committed:
Update README.md

MechComp committed:
Small fixes to license info view

michelve committed:
Merge pull request #94 from MechComp/master

Small fixes to license info view

fabolivark committed:
Remove unnecessary metadata

michelve committed:
Merge pull request #95 from ab2webco/master

Remove unnecessary metadata

ricardo777 committed:
Update wc_licenses_class.php

There are 6 TD tags and not 5

michelve committed:
Merge pull request #98 from ricardo777/patch-1

There are 6 TD tags and not 5 Update wc_licenses_class.php

michelve committed:
Update README.md

future updates notice

michelve committed:
6.1.0

michelve committed:
6.1.0 release

dev branch added - 6.1.0 release

michelve committed:
improvements

fix license creation
added language files

michelve committed:
improvements

fix: Corrected issue where domains and devices were not added to the API listener table
feat: Added license view functionality when managing license keys in wp-admin
feat: Introduced VERIFY_KEY_API constant for API key verification
feat: Enabled removal of license registrations in wp-admin
Removed: log comments func

michelve committed:
bulk license generation support

Added: bulk license generation support for existing woo orders, option must be turned on within tools
Added: SLM Tools page have a new section, for bulk license creation and database backup

michelve committed:
Update README.md

michelve committed:
improved UI and bulk generation logic

updated: updater plugin
updated: match wp ui style and classes
fixed: logic when creating bulk licenses

michelve committed:
WooCommerce enhancements

added: metabox to the woocommerce order details page, where a user can generate a new license key.
improved: validation check and logic when manually creating license keys

michelve committed:
Update slm-plus.php

incremented version

michelve committed:
release workflow

michelve committed:
Create release.yml

michelve committed:
Update release.yml

michelve committed:
Delete .github/workflows/release.yml

michelve committed:
Create buildblugin.yml

michelve committed:
added workflow action

michelve committed:
Update buildblugin.yml

michelve committed:
node 20

michelve committed:
Update buildblugin.yml

michelve committed:
update

actions/checkout@v4: Updated from v3 to v4, which is configured to use Node 20.
actions/setup-node@v4: Updated from v3 to v4, ensuring compatibility with Node 20.

michelve committed:
Update buildblugin.yml

michelve committed:
sign update

michelve committed:
Update buildblugin.yml

michelve committed:
updated

michelve committed:
Update buildblugin.yml

michelve committed:
updated

michelve committed:
Update buildblugin.yml

michelve committed:
simplified

michelve committed:
Update buildblugin.yml

michelve committed:
Update buildblugin.yml

michelve committed:
Create generate-changelog.yml

michelve committed:
Update generate-changelog.yml

michelve committed:
Update generate-changelog.yml

michelve committed:
workflow

added workflow

github-actions[bot] committed:
Update CHANGELOG.md for new release

michelve committed:
Update buildblugin.yml

michelve committed:
Update generate-changelog.yml

michelve committed:
actions

michelve committed:
Merge branch 'dev-6.1.0' of https://github.com/michelve/software-license-manager into dev-6.1.0

github-actions[bot] committed:
Update CHANGELOG.md for new release

michelve committed:
my licenses page

added my licenses and view license page under my account

michelve committed:
label and classes

added better labels for front end within table view

michelve committed:
Merge branch 'dev-6.1.0' of https://github.com/michelve/software-license-manager into dev-6.1.0

michelve committed:
updated html

michelve committed:
updated name and default values for wood product creation

Fixed: field name spelling
Fixed: default values when adding a product

michelve committed:
6.1.6 ver

version 6.1.6

michelve committed:
Updated settings and improved database upgrade

removed: bootstrap libraries, not longer needed
removed: unneeded columns from device and domain tables

github-actions[bot] committed:
Update CHANGELOG.md for new release

michelve committed:
Update README.md

michelve committed:
api endpoint improvements

improved: API endpoint
improved: expiration task handler
added: sample files for integration

michelve committed:
Merge branch 'dev-6.1.0' of https://github.com/michelve/software-license-manager into dev-6.1.0

michelve committed:
Updates 6.1.7

updated: version
fixed: when viewing a license in edit mode values would not display for user information

michelve committed:
Merge branch 'master' into dev-6.1.0

michelve committed:
Merge pull request #102 from michelve/dev-6.1.0

Dev 6.1.7 Release

michelve committed:
new images, and fixed typo

michelve committed:
6.1.7 release

github-actions[bot] committed:
Update CHANGELOG.md for new release

github-actions[bot] committed:
Update CHANGELOG.md for new release

michelve committed:
added workflows

michelve committed:
added activity log

added activity log and updated old files

michelve committed:
Merge branch 'master' of https://github.com/michelve/software-license-manager

michelve committed:
Update release-drafter.yml

michelve committed:
Update generate-changelog.yml

michelve committed:
deleted actions

michelve committed:
builder

michelve committed:
6.1.8 release

6.1.8 release

michelve committed:
builder

michelve committed:
update builder

michelve committed:
Update release.yml

michelve committed:
Update build.xml

michelve committed:
updated files

michelve committed:
Delete .github/workflows/release.yml

michelve committed:
builder

michelve committed:
Merge branch 'master' of https://github.com/michelve/software-license-manager

michelve committed:
Update build.yml

michelve committed:
Update build.yml

michelve committed:
updated builder

michelve committed:
Update build.yml

michelve committed:
Update build.yml

michelve committed:
Update build.yml

michelve committed:
Update build.yml

michelve committed:
Merge branch 'master' of https://github.com/michelve/software-license-manager

michelve committed:
Update build.yml

michelve committed:
Update build.yml

michelve committed:
Update README.md

michelve committed:
Update README.md

michelve committed:
Delete package.json

michelve committed:
6.1.9

Enhanced security measures, optimized plugin functionality, and resolved minor bugs.

michelve committed:
Add new license renewal workflow and enhanced templates

Features:

Added new license renewal workflow to track historical orders and update associated_orders column for licenses.
Introduced improved handling for WooCommerce order IDs during license renewal.
Added new templates for license management, including renewal pages and license product display suggestions for empty carts.
Improvements:

Enhanced wc_slm_create_new_license to manage expiration dates, logging, and associated orders seamlessly.
Refactored license renewal logic to ensure existing orders are appended to the associated_orders column.
Improved database schema management for adding missing columns like associated_orders, ensuring backward compatibility and smoother upgrades.
Introduced structured retrieval of associated orders with the SLM_Utility::slm_get_associated_orders function to return well-formatted data.
Fixes:

Addressed metadata dependency by replacing _slm_lic_key usage with database lookups for better accuracy and reliability.
Fixed database version handling logic to ensure schema updates are properly applied during version upgrades.
Templates:

Added license cart template with suggested license products display for empty cart scenarios.
Included robust template structure to handle renewal messages and new license purchases dynamically.
Notes:

The new renewal workflow improves traceability by appending historical orders to the license's associated_orders.
Templates are designed to integrate seamlessly with WooCommerce workflows, providing a consistent user experience.

michelve committed:
Add custom block category "SLM Plus" and improve shortcode blocks

feat: Add custom block category "SLM Plus" and improve shortcode blocks

- Registered a new block category "SLM Plus" for better organization.
- Updated "Forgot License" and "List Licenses" blocks to assign them to the custom category.
- Improved block editor previews for both blocks:
  - "Forgot License" block displays a preview with a form placeholder.
  - "List Licenses" block displays a preview with a sample list of licenses.
- Fixed issues with blocks appearing under "Uncategorized" by ensuring proper category registration.
- Ensured unique naming in block scripts to avoid conflicts.
- Added clear styles for block previews and frontend rendering in `slm-blocks.css`.
- Verified shortcode functionality for rendering licenses dynamically.

michelve committed:
Update README.md

### How to Use the New Shortcodes and Blocks

#### Shortcodes:
1. **Forgot License**: `[slm_forgot_license]`
   - Displays a form where users can enter their email to retrieve license information.
   - Can be added to any page or post manually.

2. **List Licenses**: `[slm_list_licenses]`
   - Dynamically displays a table of licenses associated with the logged-in user.
   - Useful for pages where users manage their licenses.

#### Blocks:
1. **Forgot License Block**:
   - Found under the "SLM Plus" category in the block editor.
   - Provides an interactive form preview in the editor.
   - On the frontend, renders the `[slm_forgot_license]` shortcode.

michelve committed:
Merge pull request #103 from michelve/6.2-dev

6.2 dev

michelve committed:
Update slm-plus.php

michelve committed:
Fixed: lic status table

Fix: status table not creating after update

michelve committed:
feat: Update log creation to support schema changes and enhance compatibility

- Updated the `create_log` method to handle the new `time` (DATETIME) and `time_only` (TIME) columns.
- Ensured sanitized input for `license_key`, `action`, and `origin`.
- Improved error handling with detailed success and failure logs.
- Streamlined origin determination for better readability.
- Added example usage and database insertion validation.

DAnn2012 committed:
Update slm-subscribers.php

michelve committed:
Merge pull request #108 from DAnn2012/patch-1

Fixed Function in slm-subscribers.php
